src/bosh.js

```diff
@@ -632,6 +636,9 @@ Strophe.Bosh.prototype = {
                      this._requests[0].age() > Math.floor(Strophe.SECONDARY_TIMEOUT * this.wait))) {
                     this._restartRequest(0);
                 }
+
+                this._conn.ridChange(Number(req.rid) + 1);
+
```

Comment:
> Why do you add 1 to the RID here

Because at this moment I know that the <code>req.rid</code> was valid and therefore the next valid rid is +1. If the connection aborts, I can use the new value to resume my connection. At the beginning of my project I saved the rid in <code>_buildBody</code>, but this did not work.

> why do you typecast it to a Number

The <code>req.rid</code> is of type string and +1 would result in string concatenation.